stock_picking_actual_date/models/stock_move.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2024 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo import api, fields, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    actual_date = fields.Date(
+        compute="_compute_actual_date",
+        store=True,
+    )
+
+    @api.depends("date", "picking_id.actual_date")
+    def _compute_actual_date(self):
+        for rec in self:
+            if rec.picking_id.actual_date:
+                rec.actual_date = rec.picking_id.actual_date
+                continue
+            tz = self.env.company.partner_id.tz or self.env.user.tz
+            rec.actual_date = fields.Date.context_today(
+                self.with_context(tz=tz), rec.date
+            )
```

Comment:
After some thinking, I'm now in favor of dropping the consideration on company's timezone in this module, as the assumed situation is rather peculiar.  Instead, we just add a _get_timezone() method to make it extendable.

```suggestion
    def _get_timezone(self):
        return self.env.context.get("tz") or self.env.user.tz or "UTC"

    @api.depends("date", "picking_id.actual_date")
    def _compute_actual_date(self):
        tz = self._get_timezone()
        for rec in self:
            if rec.picking_id.actual_date:
                rec.actual_date = rec.picking_id.actual_date
                continue
            rec.actual_date = fields.Date.context_today(
                self.with_context(tz=tz), rec.date
            )
```